aten/src/ATen/SparseCsrTensorImpl.cpp

```diff
@@ -239,7 +239,7 @@ IntArrayRef SparseCsrTensorImpl::strides_custom() const {
   TORCH_CHECK(false, "Sparse ", at::sparse_csr::layoutToString(layout_, /*upper=*/true), " tensors do not have strides");
 }
 SymIntArrayRef SparseCsrTensorImpl::sym_strides_custom() const {
-  TORCH_CHECK(false, "Sparse ", at::sparse_csr::layoutToString(layout_, /*upper=*/true), " tensors do not have strides");
+  return TensorImpl::sym_strides_custom(); // TODO: remove override altogether?
```

Comment:
No. Sparse tensors don't have strides (and storage_offset) by definition and returning the default strides here is wrong and meaningless. To but it differently, it is one of the methods to shoot yourself in the foot. See an alternative approach in my review summary.

Reply:
I don't disagree, but... the COO case supports strides, since elem.stride() was returning (0,0,...) when layout was coo. Is that intentional then?

(not to argue, just for my own understanging ;-)

Reply:
> but... the COO case supports strides, since elem.stride() was returning (0,0,...)

IMHO, this is a historical mistake. The purpose of the strides-storage_offset pair is to compute the locations of tensor values in memory which is not applicable for sparse tensors (read: pytorch is not ready for [GCS](https://github.com/pearu/gcs/) that would make strides meaningful for sparse tensors as well).